akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -300,6 +300,7 @@ private[stream] object VirtualProcessor {
   case object Completed extends Termination
   case class Failed(ex: Throwable) extends Termination
 
+  // FIXME this is not a legal Subscriber in the RS sense as a Subscriber may never be attached to more than 1 Publisher
```

Comment:
This is a private object precisely for the reason that it is correct only as used here, not in general. The VirtualProcessor uses this object just to atomically switch behavior in the right way, this InertSubscriber is never subscribed to any Publisher. Therefore this FIXME should be removed. If you want to use the CancellingSubscriber then please instantiate that here as a singleton in order to retain the same efficiency.